examples/vision/semantic_image_clustering.py

```diff
@@ -0,0 +1,576 @@
+"""
+Title: Semantic Image Clustering
+Author: [Khalid Salama](https://www.linkedin.com/in/khalid-salama-24403144/)
+Date created: 2021/02/28
+Last modified: 2021/02/28
+Description: Semantic Clustering by Adopting Nearest neighbors (SCAN) algorithm.
```

Comment:
Suggestion:

- State the task the algorithm tries to solve - in this case it's image classification (I had to read the paper to understand what SCAN was)

```suggestion
Description: Image classification on CIFAR-10 with Semantic Clustering by Adopting Nearest neighbors (SCAN).
```

(the description is sort of similar to https://keras.io/examples/vision/image_classification_from_scratch/)

Reply:
Strictly speaking, it's not classification, it's clustering, as the target 'classes' are considered unknown.